### PR TITLE
update the ubuntu version

### PR DESCRIPTION
**Pull request by ArshErgon**
Fixes: #1689
updated the ubuntu-version from 20.04 to 22.04

Thank you for making Rust GCC better!

If your PR fixes an issue, you can add "Fixes #issue_number" into this
PR description and the git commit message. This way the issue will be
automatically closed when your PR is merged. If your change addresses
an issue but does not fully fix it please mark it as "Addresses #issue_number"
in the git commit message.

Here is a checklist to help you with your PR.

- \[ ] GCC development requires copyright assignment or the Developer's Certificate of Origin sign-off, see https://gcc.gnu.org/contribute.html or https://gcc.gnu.org/dco.html
- \[ ] Read contributing guidlines
- \[ ] `make check-rust` passes locally
- \[ ] Run `clang-format`
- \[ ] Added any relevant test cases to `gcc/testsuite/rust/`

Note that you can skip the above if you are just opening a WIP PR in
order to get feedback.
---

*Please write a comment explaining your change. This is the message
that will be part of the merge commit.

### CODE REVIEW

**Review by CohenArthur — CHANGES_REQUESTED** [review]
Thank you for taking a crack at this! One of the steps mentioned in #1689 is really important, and that's what causing CI to fail for this PR. We need to update the "expected warnings" list, which is being check in the `Check for new warnings` step in the ccpp.yml workflow. If you're interested in working on this, I would suggest taking a look at #1026

File: .github/workflows/ccpp.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ jobs:
            cd gccrs-build;
            ../configure \
                --enable-languages=rust \
-               --disable-bootstrap \
+               --enable-bootstrap \
```
Comment (CohenArthur) [inline .github/workflows/ccpp.yml:50]: This needs to stay as `--disable-bootstrap`

File: .github/bors_log_expected_warnings [hunk 2]
```diff
@@ -148,3 +148,125 @@ gengtype-lex.cc:357:15: warning: this statement may fall through [-Wimplicit-fal
...
+-+../../gcc/lto-wrapper.cc:1365:10: warning: format not a string literal and no format arguments [-Wformat-security]
+-+../../gcc/lto-wrapper.cc:1957:23: warning: format not a string literal and no format arguments [-Wformat-security]
+-+../../gcc/tree-ssa-sccvn.cc:280:21: warning: format not a string literal and no format arguments [-Wformat-security]
+-+../../libcpp/expr.cc:821:35: warning: format not a string literal and no format arguments [-Wformat-security]
+-+../../libcpp/expr.cc:824:38: warning: format not a string literal and no format arguments [-Wformat-security]
+-+../../libcpp/expr.cc:834:33: warning: format not a string literal and no format arguments [-Wformat-security]
+-+../../libcpp/macro.cc:185:26: warning: format not a string literal and no format arguments [-Wformat-security]
+-+../../libcpp/macro.cc:214:34: warning: format not a string literal and no format arguments [-Wformat-security]
+-+../../libcpp/macro.cc:3704:25: warning: format not a string literal and no format arguments [-Wformat-security]
+-+../../libcpp/macro.cc:3719:25: warning: format not a string literal and no format arguments [-Wformat-security]
```
Comment (CohenArthur) [inline .github/bors_log_expected_warnings:272]: You need a newline at the end of this file

**Review by CohenArthur — COMMENTED** [review]
I think this looks good @ArshErgon, I believe that adding the newline at the end of the expected_warnings file will fix the remaining CI failure

File: .github/workflows/ccpp.yml (hunk 1 above)
Comment (ArshErgon) [reply .github/workflows/ccpp.yml]: Yeah, I resolve this, thanks

File: .github/bors_log_expected_warnings (hunk 2 above)
Comment (ArshErgon) [reply .github/bors_log_expected_warnings]: thanks I was stuck on this one.

**Comment by CohenArthur** [comment]
@ArshErgon do you want some help to move this forward? I can take care of the missing warnings if you'd like

**Comment by ArshErgon** [comment]
> @ArshErgon do you want some help to move this forward? I can take care of the missing warnings if you'd like

yeah, I don't know what's getting wrong?

**Comment by CohenArthur** [comment]
@ArshErgon I've updated the PR and rebased it into a single commit for which you are the author. Let me know if you're happy with that and we can merge this :)

**Pull request merged**